### PR TITLE
Refurbish turret control device

### PR DESCRIPTION
**Pull request by ghost**
Turret control devices (that weird thing you use to turn the turrets on or off in AI Upload and Tcomms) were in need of a bit of love, I feel. I'm here to give them that love. Should be no changes on anything else

- Changed the sprite to use the airlock control one. It still keeps the green light/yellow light/red light system, but at least it looks a bit fancier and it has a card reader + buttons
- Some immersive descriptions, touching up the UI wording a bit to be less heavy
- Turret control now uses adjacency, so you don't have to stand under it to activate it. It means you can hit it from a tile away now, but infinitely better than hunting for the "origin tile" of that fucker
- Future proofing the code so adding an instance of the turret control set to deactivated/lethal by default actually works properly out of the box (turrets and sprite will be updated)
- Added Alt Click for normal mobs (turns the turrets on/off assuming it is unlocked) and Shift/Ctrl/Alt click for AI mirroring the "logic" of airlocks
- Added a bunch of visible messages to using the device

Balance change : 

- Emagging the turret control now additionally breaks its "threat indicator". To put it more simply, it will always show as "turrets are deactivated". Can be used for a nasty surprise by crafty Traitors (you can still turn the turrets on/off as normal, only the sprite is 'broken'. It will also be impossible to lock as previously)

New sprite, showing stun variant of the new turret controller sprite : 

turrent

:cl:
 * rscadd: Turret control has a new sprite and can now be activated from a tile away. Emagging the turret control will now break the "threat indicator" aka the blinky colored light, showing the turrets as deactivated for tator traps.
* tweak. Added alt-click shortcut for normal mobs to turn the turrets on a turret control device on/off. Added Shift/Alt/Ctrl AI shortcuts to turret control devices. Shift click enables/disables the turrets, Alt click turns the turrets to stun/lethal. Ctrl click locks the interface

### CODE REVIEW

**Comment by brndd** [comment]
Make it so you can alt-click them to turn the turrets on and off as long as you have access.

**Comment by ghost** [comment]
As AI/Silicons ?

**Comment by juanototo2** [comment]
I would screech about something something unatomic please do not bundle balance changes but then I realized it's not the end of the world and it's technically better than baguloose simulator 2019

**Comment by brndd** [comment]
>As AI/Silicons ?

No, as anyone. It's pretty annoying having to open UIs to do common single-click tasks like disabling the turrets. Alt-clicking is used for this purpose on many devices these days, for instance gas pumps.

File: code/game/machinery/turrets.dm [hunk 1]
```diff
@@ -288,6 +288,31 @@
 		src.die()
 	return
 
+/obj/machinery/turret/attack_animal(mob/living/simple_animal/M as mob)
```
Comment (ShiftyRail) [inline code/game/machinery/turrets.dm]: As mob isn't needed.

File: code/game/machinery/turrets.dm [hunk 2]
```diff
@@ -288,6 +288,31 @@
...
+		if (src.health <= 0)
+			src.die()
+	else
+		to_chat(M, "<span class='warning'>That object is useless to you.</span>")
+
+/obj/machinery/turret/attack_alien(mob/living/carbon/alien/humanoid/M as mob)
+	if(!(stat & BROKEN))
+		M.do_attack_animation(src, M)
+		playsound(src, 'sound/weapons/slash.ogg', 25, 1, -1)
+		visible_message("<span class='danger'>[] has slashed at []!</span>", M, src)
```
Comment (ShiftyRail) [inline code/game/machinery/turrets.dm]: My eyes
`visible_message("<span class='danger'>[M] has slashed at [src]!</span>")` instead, please

File: code/game/machinery/turrets.dm [hunk 3]
```diff
@@ -336,157 +362,122 @@
...
 	user.set_machine(src)
+	// Oh god what
 	var/loc = src.loc
-	if (istype(loc, /turf))
+	if(istype(loc, /turf))
 		loc = loc:loc
-	if (!istype(loc, /area))
-		to_chat(user, text("Turret control badly positioned - loc.loc is [].", loc))
+	if(!istype(loc, /area))
+		to_chat(user, "Turret control badly positioned - loc.loc is [loc].") //Debug message
```
Comment (ShiftyRail) [inline code/game/machinery/turrets.dm]: Make it something a bit more IC. `Turret control must be placed on a tile.`
Also, I _think_ this whole block simply checks if the controller is placed on a turf and if that turf isn't in space

File: code/game/machinery/turrets.dm (hunk 2 above)
Comment (ghost) [reply code/game/machinery/turrets.dm]: That wasn't part of the changes, just displacing this code block since it's regular turret stuff, but sure

**Comment by ghost** [comment]
Updated the PR and PR message, recheck

**Comment by brndd** [comment]
Good work on the shortcuts.

**Pull request merged**